src/librustc/ty/layout.rs

```diff
@@ -1514,28 +1513,28 @@ impl<'a, 'tcx> LayoutOf for LayoutCx<'tcx, ty::maps::TyCtxtAt<'a, 'tcx, 'tcx>> {
 }
 
 // Helper (inherent) `layout_of` methods to avoid pushing `LayoutCx` to users.
-impl<'a, 'tcx> TyCtxt<'a, 'tcx, 'tcx> {
+impl<'a, 'tcx, 'empty> TyCtxt<'a, 'tcx, 'empty> {
```

Comment:
Please use `impl<'a, 'tcx> TyCtxt<'a, 'tcx, '_>` if you don't want to name the lifetime.

Reply:
I tried that, that doesn't compile

Reply:
Doesn't it need a feature-gate? `in_band_lifetimes` I think?